### DISCUSSION TITLE
No Such Module '3RDPARTYSDK' error  & some question about project structure

### DISCUSSION THREAD

**Discussion by HTCZeus** [callstack/react-native-builder-bob#444] [Q&A] [answered]
Hi, first of all I am not a experienced developer,
I have developed a react-native bridge for iOS and android SDK. I have tested it in react-native project they are working fine but I want to publish my native modules (IOS and Android modules) as a npm package so everyone can use it without coding swift or java. So I found out builder bob. I created new project with npx create-react-native-library@latest react-native-mysdk
I have moved my IOS codes to IOS folder in the root. But my iOS native codes needs 3rd party pods to work. Previously I have added these pods to my pod file in my react-native project but what should I do now ? Should I create a pod file in the root iOS folder or should I add my pods to podspec file with s.dependency  ? I am really confused 

I tried to add these pods to example>ıos>pod file but I got a error when building => No such module '3RDPARTYSDK'

**Comment by atlj** [comment, marked as answer]
You can use the `YOUR_MODULE_NAME.podspec` file to add 3rd party dependencies in this case.

Please take a look at this answer: https://github.com/callstack/react-native-builder-bob/discussions/430#discussioncomment-6579793
